packages/manager/modules/overthebox/src/overthebox/autoconfigure/dns/template.html

```diff
@@ -0,0 +1,248 @@
+<div>
+    <oui-message
+        data-ng-if="$ctrl.successMessage"
+        type="success"
+        aria-close-button-label="{{:: 'overTheBox_autoconfigure_close' | translate}}"
+        dismissable
+    >
+        <span data-ng-bind="$ctrl.successMessage"></span>
+    </oui-message>
+    <oui-message
+        data-ng-if="$ctrl.errorMessage"
+        type="error"
+        aria-close-button-label="{{:: 'overTheBox_autoconfigure_close' | translate}}"
+        on-dismiss="$ctrl.onDismiss()"
+        dismissable
+    >
+        <span data-ng-bind="$ctrl.errorMessage"></span>
+    </oui-message>
+    <div class="oui-box oui-box_light">
+        <h4
+            class="oui-box__heading"
+            data-translate="overTheBox_autoconfigure_dns_server"
+        ></h4>
+        <div data-ng-if="$ctrl.nameServers.length === 0" class="mb-2">
```

Comment:
A common mistake is to manipulate data that is not yet defined until the api returns it. Angular catches it so it does not throw but it is good practice to 1. take care of undefined / null variables 2. avoid code repetition (L24 & L78) 
```suggestion
        <div data-ng-if="$ctrl.hasNoNameServers" class="mb-2">
```
```javascript
get hasNoNameServers() {
  return this.nameServers?.length === 0;
}
```
See also for `localDomains` L129 & L197